src/main/kotlin/no/nav/k9/los/integrasjon/abac/PepClient.kt

```diff
@@ -268,50 +172,69 @@ class PepClient constructor(
     }
 
     override suspend fun harTilgangTilOppgave(oppgave: Oppgave): Boolean {
-        return !oppgave.harFagSaksNummer() || harTilgangTilLesSak(
-            fagsakNummer = oppgave.fagsakSaksnummer,
-            aktørid = oppgave.aktorId
-        )
+        return harTilgang("k9sak", Action.read, oppgave.fagsakSaksnummer, oppgave.aktorId, false)
     }
 
-    override fun harTilgangTilOppgaveV3(oppgave: no.nav.k9.los.nyoppgavestyring.visningoguttrekk.Oppgave, bruker: Saksbehandler): Boolean {
-         if (oppgave.hentVerdi("saksnummer") == null) {
-             return true
-         } else return runBlocking {
-             harTilgangTilLesSak(oppgave.hentVerdi("saksnummer")!!, oppgave.hentVerdi("aktorId")!!, bruker)
-         }
+    override suspend fun harTilgangTilOppgaveV3(
+        oppgave: no.nav.k9.los.nyoppgavestyring.visningoguttrekk.Oppgave,
+        action: Action,
+        auditlogging: Boolean
+    ): Boolean {
+        return harTilgang(oppgave.oppgavetype.eksternId, action, oppgave.hentVerdi("saksnummer"), oppgave.hentVerdi("aktorId"), auditlogging)
     }
 
-    override suspend fun harTilgangTilÅReservereOppgave(oppgave: no.nav.k9.los.nyoppgavestyring.visningoguttrekk.Oppgave, bruker: Saksbehandler) : Boolean {
-        val requestBuilder = when (oppgave.oppgavetype.eksternId) {
-            "k9sak" -> {
-                XacmlRequestBuilder()
-                    .addEnvironmentAttribute(ENVIRONMENT_PEP_ID, "srvk9los")
-                    .addResourceAttribute(RESOURCE_DOMENE, DOMENE)
-                    .addResourceAttribute(RESOURCE_TYPE, TILGANG_SAK)
-                    .addActionAttribute(ACTION_ID, "update")
-                    .addResourceAttribute("no.nav.abac.attributter.resource.k9.behandlings_uuid", oppgave.eksternId)  //TODO los skal ikke kjenne til denne detaljen. Oppgavetype må utvides med attributtreferanse
-                    .addAccessSubjectAttribute(SUBJECT_TYPE, INTERNBRUKER)
-                    .addAccessSubjectAttribute(SUBJECTID, bruker.brukerIdent!!)
-            }
-            "k9klage" -> {
-                XacmlRequestBuilder()
+    private suspend fun harTilgang(
+        oppgavetype: String,
+        action: Action,
+        saksnummer: String?,
+        aktorId: String?,
+        auditlogging: Boolean
```

Comment:
"auditLogging" bør antagelig være en enum med 3 verdier:

IKKE_LOGG  (tilfeller hvor 'false' i dag)
ALLTID_LOGG (normal bruk av løsningen, utenom under)
LOGG_VED_PERMIT (plukk neste fra kø, vis neste i køer - i disse tilfellene ber bruker om å få neste som hen har tilgang på, jeg tenker det blir rart (uheldig for saksbehandler) om vi logger DENY på de som vi hopper over)

Reply:
Veldig enig i forslaget, men ALLTID_LOGG er kanskje heller unntaket enn normal bruk i løsningen? For alt bortsett fra søk direkte på saksnummer/fnr/journalpost filtreres oppgavene bort i stillhet, og det er fort gjort å få unødvendig mange DENY i auditlog. F.eks er det ingen garanti for at pepcache klarer å fange opp alle oppgavene allerede i db, og da vil pepclient filtrere dem bort og samtidig logge DENY.

Reply:
Laget enumen med de verdiene du listet opp.

Dett er de tre stedene der metoden harTilgangTilOppgaveV3 brukes:
* OppgaveKoTjeneste.hentOppgaverFraKø. Auditlogging LOGG_VED_PERMIT.
* OppgaveQueryService.mapAktivOppgave. Auditlogging LOGG_VED_PERMIT.
* ReservasjonV3Tjeneste.sjekkTilganger. Auditlogging IKKE_LOGG.

ALLTID_LOGG er dermed ikke i bruk.